### ISSUE TITLE
Communicating between different instances pretty slow

### ISSUE THREAD

**Issue by keganlester** [laravel/valet#908]
I am kind of new to networking so this might be a dumb question but I am working with a custom API locally and the communication/requests between one site to the API are slow. 

for example, I make a get request to api.test from mysite.test and it isn't that speedy for the small amount of data. I tried securing the domain and I got a curl error 60 with a certificate problem.

Is there a correct way to set this up or should I just expect this to be slow?

**Comment by drbyte** [comment]
I would expect things to be speedy.

But if it's painfully slow then perhaps you're running into timeouts while DNS lookups failover from provider to provider.

In this case, consider adding `127.0.0.1` to your System Preferences->Network->Advanced->DNS, as the first (or even your "only") choice:

<img width="351" alt="Screen Shot 2020-02-12 at 5 51 56 PM" src="https://user-images.githubusercontent.com/404472/74384973-729a5d00-4dc0-11ea-8d3c-d98042a13b8f.png">


If command-line curl is giving you a certificate error, you could create a `~/.curlrc` file containing Valet's Certificate Authority details:
```text
cacert=/Users/YOUR_USERNAME/.config/valet/CA/LaravelValetCASelfSigned.pem
```

Then again, it depends on how are you making the API get request. You didn't explain what packages/code/subsystems are being told to make the API call.

**Comment by keganlester** [comment]
@drbyte I have the DNS to prioritize 127.0.0.1 first and I am using a simple guzzle http client (v6.3)

**Comment by drbyte** [comment]
The tiny note at the end of the Guzzle quickstart docs may give a clue to the `curl error 60 with a certificate problem`:
http://docs.guzzlephp.org/en/stable/quickstart.html#environment-variables

Let's try this:
- create a new file: `/usr/local/etc/php/7.4/conf.d/php-openssl-ca.ini`
 (where 7.4 is your PHP version)

- with the following contents:
(replacing YOUR_USERNAME with your Mac user's Home directory name)
```conf
; This allows PHP fopen or guzzlehttp to recognize Valet's CA

[openssl]
openssl.cafile="/Users/YOUR_USERNAME/.config/valet/CA/LaravelValetCASelfSigned.pem"
```

- `valet restart php`

**Comment by keganlester** [comment]
@drbyte Thank you for the help by the way! I am getting that the domain doesn't exist. Before to fix this I added it to the top of my DNS but it looks like thats not working anymore.

**Comment by drbyte** [comment]
> I am getting that the domain doesn't exist

Hmmm. 

You haven't mentioned yet whether you've been able to open both sites in your browser, or at least get a response from Postman etc.
(I know you said response is "slow", but you didn't say that the slow responses were "successful", so maybe the site just isn't responding either?)

**Comment by keganlester** [comment]
@drbyte I can load everything in chrome, I can not load anything in postman. That is very weird.

**Comment by drbyte** [comment]
Weird.

Thoughts:

If Postman is saying domain doesn't exist, then I'm guessing it's somehow using a different DNS resolver than what your core OS is set to use. Have you told it to use a proxy of some sort? 

If Postman is saying the domain exists but has a certificate error, then it's not trusting the certificates, which means it's using a different certificate authority than your OS is configured to trust.

**Comment by keganlester** [comment]
@drbyte yes you are correct on our company network we have other configuration settings for DNS. New to this whole networking thing I have to admit.

**Comment by drbyte** [comment]
Well, if `127.0.0.1` is first in your list of DNS resolvers (screenshot above) then I'd expect Valet to be responding "first", so it should both "work" and be "speedy".

If you've got another IP listed first (or if something has been done to interfere with Valet's priority), then that might explain the delays or failures.

The default Valet config also allows your `hosts` file to do its normal thing, so if you've added stuff there for this domain you might want to comment it out.

Aside: I'm assuming ALL these domains you're trying to access ARE on YOUR machine, right?

**Comment by keganlester** [comment]
@drbyte yes all on the machine. I moved from running like 10 docker containers to using valet so switching to this and managing DNS is a littl difficult. But yes the DNS has 127.0.0.1 first.

**Comment by drbyte** [comment]
I'm thinking it'd be useful to try to narrow the scope, so I'm curious about:
Can other browsers access these sites without speed delays?

I'm also curious if your company network's administrator knows anything about special configurations that have been applied?

**Comment by driesvints** [comment]
Hi there, we're cleaning up the Valet issue tracker and are moving all feature requests/support questions to the Discussions tab. We'll be using the issue tracker solely for bugs with Valet from now on. You're welcome to continue the discussion on in the Discussions tab. Thanks!
